src/binder/utils.cpp

```diff
@@ -267,14 +267,17 @@ namespace BINDER_SPACE
 
         if (!path.IsEmpty())
         {
-            WCHAR wszCanonicalPath[MAX_LONGPATH];
+            PathString wszCanonicalPathString;
+            WCHAR * pwszCanonicalPath = wszCanonicalPathString.OpenUnicodeBuffer(MAX_LONGPATH);
             PlatformPath(path);
 
             // This is also defined in rotor pal
-            if (PathCanonicalizeW(wszCanonicalPath, path))
+            if (PathCanonicalizeW(pwszCanonicalPath, path))
```

Comment:
The contract for `PathCanonicalize` is hard code to `MAX_PATH`. The right fix here is to delete the `BINDER_SPACE::PathCanonizalize` method and its uses - without replacing them with anything. I have reviewed all uses - none of them is required.